### PR TITLE
Replace custom logger with stdlib logging

### PR DESCRIPTION
**Pull request by Wenzel**
Replace custom logger.py with python stdlib logging library

### CODE REVIEW

**Comment by il-steffen** [comment]
Thanks a lot!


Comparing some old/new log output:
`0:00:39.750624 Worker-12 exit code: None` => `INFO:kafl_fuzzer.worker.qemu:exit code: None`

- the file log lost the full time stamp (time since fuzzer start) - I think we should keep this since one often considers fuzzer activity over longer time
- qemu still generates its own Worker-N prefix - maybe we hand it the logger created by Worker?
- the "worker-N" prefix got lost when changing the `logger.file_log()` in qemu.shutdown()
- lets rename fuzzer.log to kafl_fuzzer.log? "fuzzer" is too generic, "kafl_fuzzer" maps to the python name.

The level+module prefix is nice but unfortunately destroys alignment, making it really hard to find something useful in the log:
```
DEBUG:kafl_fuzzer.manager.manager:Received new input (exit=regular): b'\xe4Gh)q]\x80]d\x13FU\xee\xa3\xc9\xfb\x9f\xe6\xd1\xccQ\x9c\x0e:'
DEBUG:kafl_fuzzer.worker.worker:767061:Worker-34:Stage initial found alternative payload for node 24
DEBUG:kafl_fuzzer.worker.worker:767061:Worker-34:Stage initial found alternative payload for node 56
DEBUG:kafl_fuzzer.technique.trim:stream_extend: detected 65567 padding bytes
DEBUG:kafl_fuzzer.technique.trim:stream_extend: dropped funky NUL padding (len=65567, other finds=0)
DEBUG:kafl_fuzzer.technique.trim:stream_extend: detected 64831 padding bytes
DEBUG:kafl_fuzzer.manager.manager:Received new input (exit=timeout): b'\xcd\xa2\xb1\x05\xe6o\xbb5\xcex\xcf\x0c\xc4=\xd2\xd8\xe1=\xffH\xe4\xde\xb4A'
DEBUG:kafl_fuzzer.manager.manager:Received new input (exit=regular): b'\xcd\xa2\xb1\x05\xe6o\xbb5\xcex\xcf\x0c\xc4=\xd2\xd8\xe1=\xffH\xe4\xde\xb4A'
```
```
0:00:57.640407 Received new input (exit=regular): b'\xc3\xfb\xb0,26\x1a\x95c\xf1\xaeNi\x8f\xbd\xde\x8c\x8d\x081\xdcx\x04('
0:00:57.640693 Received duplicate payload with exit=regular, discarding.
0:00:57.665143 stream_extend: detected 65567 padding bytes
0:00:57.705712 Received new input (exit=regular): b'\xc3\xfb\xb0,26\x1a\x95c\xf1\xaeNi\x8f\xbd\xde\x8c\x8d\x081\xdcx\x04('
0:00:57.718928 Worker-03 HAVOC times: afl: 7.8, splice: 19.2, grim: 0.0, rdmsa: 0.0
0:00:57.721715 Worker-62 HAVOC times: afl: 7.4, splice: 18.6, grim: 0.0, rdmsa: 0.0
0:00:57.730822 Worker-62 RedQ-Dict: Have performed 0 iters
0:00:57.733465 Worker-01 HAVOC times: afl: 7.9, splice: 19.5, grim: 0.0, rdmsa: 0.0
0:00:57.766072 Worker-01 Stage initial found alternative payload for node 39
```

I think we want to use names based on subsystems, not module hierarchy, and append them to the Worker-N notation for alignment. Similar to shown in the last few lines here. This is not totally consistent in the code right now, but we can fix it some other time. (Some modules need to be turned into a class first so that we can properly hand down a logger module to them.)

One more thing - maybe we want to restore console output to normal verbosity once --log is provided?

**Comment by Wenzel** [comment]
Thanks for the review !

TODO overview
- [ ] rebase PR after #16 merge
- [ ] rename `fuzzer.log` to `kafl_fuzzer.log`
- [ ] add timestamp prefix in file log
- [ ] `qemu.shutdown()`:  `worker-N` prefix git lost
- [ ] qemu still generates its own Worker-N prefix - maybe we hand it the logger created by Worker?
- [ ] file log format breaks alignment and makes it hard to follow. Rethink file log output
- [ ] restore console output to normal if `--log`

**Comment by Wenzel** [comment]
@il-steffen regarding the "Worker-N" prefix, I moved the `WorkerLogAdapter` into `common/logger.py`, and used it in [`worker.py`](https://github.com/IntelLabs/kafl.fuzzer/pull/17/files#diff-4d1b1e797bed6f6f8d7f811a9c2c15c2496ce420432ddeb816d0fdd5d2801addR43) and [`qemu.py`](https://github.com/IntelLabs/kafl.fuzzer/pull/17/files#diff-7a0f10a7fc82e304c578ab476135e79b57aef07bc86b689de49686f18b61e799R49) this time.
(before your comment, I only updated `worker.py`.

Is this output the one you expect ? (in terms of Worker-xx prefix)
image

**Comment by il-steffen** [comment]
Yes that is how it should look. I had the same `exit code` line above from the old logger out:
`0:00:39.750624 Worker-12 exit code: None`

I would avoid the extra colon separators as well, or put a space after colon.

**Review by il-steffen — APPROVED** [review]
